test/unit/client/windshaft.test.js

```diff
@@ -1,3 +1,138 @@
+import Windshaft from '../../../src/client/windshaft';
+import Viz from '../../../src/Viz';
+
+const base = {
+    _apiKey: '1234',
+    _username: 'wadus',
+    _serverURL: 'https://wadus.carto.com',
+    _getFromClause: () => 'table_name'
+};
+
+const vizMock = `
+    width: 4
+    color: ramp(linear(abs($accel_lateral), 0, 1), temps)
+    strokeWidth: 0
+    filter: $lap == 24 and animation(linear($date), 20, fade(0, 1))
+`;
+
+const MNSMock = {
+    'accel_lateral': [
+        {
+            'type': 'unaggregated'
+        }
+    ],
+    'lap': [
+        {
+            'type': 'unaggregated'
+        }
+    ],
+    'date': [
+        {
+            'type': 'unaggregated'
+        },
+        {
+            'type': 'unaggregated'
+        },
+        {
+            'type': 'unaggregated'
+        }
+    ],
+    'cartodb_id': [
+        {
+            'type': 'unaggregated'
+        }
+    ]
+};
+
+const resolutionMock = 1;
+
+const filteringMock = {
+    'preaggregation': {
+        'type': 'equals',
+        'left': {
+            'type': 'property',
+            'property': 'lap'
+        },
+        'right': {
+            'type': 'value',
+            'value': 24
+        }
+    },
+    'aggregation': {}
+};
+
+const vizInfoMock = {
+    'MNS': MNSMock,
+    'resolution': resolutionMock,
+    'filtering': filteringMock
+};
+
+const metadataMock = {
+    'properties': [],
+    'backendFiltersApplied': true
+};
+
+const urlTemplatesMock = 'url_templates_mock';
+
+const instantiationDataMock = {
+    'MNS': MNSMock,
+    'resolution': resolutionMock,
+    'filtering': filteringMock,
+    'metadata': metadataMock,
+    'urlTemplates': urlTemplatesMock
+};
+
+// TODO: add more tests to cover all Windshaft functions
 
 describe('src/client/windshaft', () => {
+    let client;
+
+    beforeEach(() => {
+        client = new Windshaft(base);
+        // Mock fetch function
+        client._getInstantiationPromise = () => ({
+            metadata: metadataMock,
+            urlTemplates: urlTemplatesMock
+        });
+    });
+
+    describe('_getServerInfoFrom', () => {
+        it('should convert viz into vizInfo', () => {
+            const viz = new Viz(vizMock);
+            const vizInfo = client._getServerInfoFrom(viz);
+            expect(vizInfo).toEqual(vizInfoMock);
+        });
+    });
+
+    describe('_needToInstantiateMap', () => {
+        it('should detect that it requires instantiation', () => {
```

Comment:
It would be nice to test again, so we see no need for a new instantiation

Reply:
And maybe a third change, altering one of the relevant properties. So it needs a new inst. again... Something so understand better the implement. from the test